packages/flutter/lib/src/animation/tween.dart

```diff
@@ -29,6 +32,14 @@ abstract class Animatable<T> {
   /// const constructors so that they can be used in const expressions.
   const Animatable();
 
+  /// Create a new [Animatable] from the provided [callback].
+  ///
+  /// See also:
+  ///
+  ///  * [Animation.drive], which provides an example for how this can be
+  ///    used.
+  const factory Animatable.fromCallback(AnimatableCallback<T> callback) = _CallbackAnimatable<T>;
```

Comment:
Just a _little_ bit more API. It felt a bit silly to create a class just to wrap a callback in the sample code